compute/metadata/metadata.go

```diff
@@ -111,6 +111,7 @@ var (
 )
 
 // OnGCE reports whether this process is running on Google Compute Engine.
```

Comment:
maybe instead of calling out specific products we can just rephrase to "... Google Compute Platforms.". This way the docs can stay up-to-date over time.

Reply:
sure, the idea is to just give more context to the function for others

going to push the change now